cmstestsuite/programstarter.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+# Contest Management System - http://cms-dev.github.io/
+# Copyright © 2012 Bernard Blackham <bernard@largestprime.net>
+# Copyright © 2013-2016 Stefano Maggiolo <s.maggiolo@gmail.com>
+# Copyright © 2013-2014 Luca Wehrstedt <luca.wehrstedt@gmail.com>
+# Copyright © 2014 Luca Versari <veluca93@gmail.com>
+# Copyright © 2014 William Di Luigi <williamdiluigi@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import atexit
+import errno
+import io
+import json
+import logging
+import os
+import signal
+import socket
+import subprocess
+import threading
+import time
+from urlparse import urlsplit
+
+from cmstestsuite import CONFIG, FrameworkException, get_cms_config
+
+
+logger = logging.getLogger(__name__)
+
+
+# Maximum number of attempts to check if a service becomes healthy.
+_MAX_ATTEMPTS = 20
```

Comment:
Maybe, but I'm using it in two classes and you can argue is a constant of the module.